support/metrics/metricsTracker.go

```diff
@@ -304,3 +311,41 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+// RegisterHandler adds an internal handler.
+func (mt *MetricsTracker) RegisterHandler(handler api.TradeMetricsHandler) {
```

Comment:
we don't need any of these functions here.

instead inside sendUpdate we can call `ComputeMetrics` on the `tradeMetricsHandler` instance that we hold and merge that output with the update map that we are sending to `sendEvent` -- that's it!